packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart

```diff
@@ -829,55 +830,69 @@ class _InspectorTreeState extends State<InspectorTree>
 //    );
 //  }
 
-  /// Compute the goal x scroll given a y scroll value.
-  ///
-  /// This enables animating the x scroll as the y scroll changes which helps
-  /// keep the relevant content in view while scrolling a large list.
-  double _computeTargetX(double y) {
-    final treeControllerLocal = treeController!;
-    final rowIndex = treeControllerLocal.getRowIndex(y);
-    double? requiredOffset;
-    double minOffset = double.infinity;
-    // TODO(jacobr): use maxOffset as well to better handle the case where the
-    // previous row has a significantly larger indent.
-
-    // TODO(jacobr): if the first or last row is only partially visible, tween
-    // between its indent and the next row to more smoothly change the target x
-    // as the y coordinate changes.
-    if (rowIndex == treeControllerLocal.numRows) {
-      return 0;
-    }
-    final endY = y += safeViewportHeight;
-    for (int i = rowIndex; i < treeControllerLocal.numRows; i++) {
-      final rowY = treeControllerLocal.getRowY(i);
-      if (rowY >= endY) break;
-
-      final row = treeControllerLocal.getCachedRow(i);
-      if (row == null) continue;
-      final rowOffset = treeControllerLocal.getRowOffset(i);
-      if (row.isSelected) {
-        requiredOffset = rowOffset;
-      }
-      minOffset = min(minOffset, rowOffset);
-    }
-    if (requiredOffset == null) {
-      return minOffset;
-    }
-
-    return minOffset;
-  }
-
   @override
   Future<void> scrollToRect(Rect rect) async {
     if (rect == _currentAnimateTarget) {
       // We are in the middle of an animation to this exact rectangle.
       return;
     }
-    _currentAnimateTarget = rect;
-    final targetY = _computeTargetOffsetY(
+
+    final initialX = rect.left;
+    final initialY = rect.top;
+    final yOffsetAtViewportTop = _scrollControllerY.hasClients
+        ? _scrollControllerY.offset
+        : _scrollControllerY.initialScrollOffset;
+    final xOffsetAtViewportLeft = _scrollControllerX.hasClients
+        ? _scrollControllerX.offset
+        : _scrollControllerX.initialScrollOffset;
+
+    final viewPortInScrollControllerSpace = Rect.fromLTWH(
+      xOffsetAtViewportLeft,
+      yOffsetAtViewportTop,
+      safeViewportWidth,
+      safeViewportHeight,
+    );
+
+    final rowIndex = treeController!.getRowIndex(rect.top);
+
+    final row = treeController!.getCachedRow(rowIndex);
+    final node = row?.node;
+    final textPreviewJson = node?.diagnostic?.json['textPreview'];
+    final textDescription = node?.diagnostic?.description;
+    final descriptions = <String>[
+      if (textDescription != null) textDescription,
+      if (textPreviewJson != null) '$textPreviewJson',
+    ];
```

Comment:
this seems fragile. What if we change how inspector nodes are displayed at some point? Is this display logic duplicated somewhere else that we could create a helper for?

Reply:
Good call on this, it is definitely fragile.
I believe in order to achieve parity we would have to decouple this aspect.
https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/inspector/diagnostics.dart#L61

I can definitely look into what it would take to do so

Reply:
+1 on this being very fragile. Add a helper function + a unittest that fails if the predicted width and the actual width f rendering a diagnostic description go out of sync for some example input.